### PR TITLE
Add fake rootfs service

### PR DESCRIPTION
**Pull request by mozhuli**
Signed-off-by: mozhuli <21621232@zju.edu.cn>
ref:  #20

### CODE REVIEW

**Comment by mozhuli** [comment]
have signed a CLA

**Comment by mozhuli** [comment]
I signed it!

File: pkg/server/testing/fake_rootfs_client.go [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+	f.Lock()
+	defer f.Unlock()
+	f.errors = make(map[string]error)
+}
+
+func resolveDiffID(digest.Digest) digest.Digest {
+	return digest.Digest("")
+}
+
+func generateChainID(layers []*descriptor.Descriptor) digest.Digest {
```
Comment (Random-Liu) [inline pkg/server/testing/fake_rootfs_client.go]: Let's use layer digest as diffID for now, and add some comment to explain that we should be using diffID.
https://github.com/opencontainers/image-spec/blob/eb02c88ef382324a1572fa8c66f836a00d865a3f/config.md#layer-diffid

**Comment by Random-Liu** [comment]
@mozhuli And please sign the CLA.

**Comment by mozhuli** [comment]
I signed it!@k8s-ci-robot

**Comment by Random-Liu** [comment]
Close and reopen to trigger the test again.

**Comment by Random-Liu** [comment]
@mozhuli Have you associate your CLA with your github account?

**Comment by mozhuli** [comment]
@Random-Liu I have associate my CLA with my github account, and the email also the same

**Comment by Random-Liu** [comment]
@mozhuli Weird...Could you rebase your PR first?

BTW, containerd will change their api https://github.com/containerd/containerd/pull/849.

Anyway, we'll be using current api for a while until their api is relatively stabilized, so this PR is still pretty useful.

**Comment by mozhuli** [comment]
I have signed the CLA

**Comment by Random-Liu** [comment]
@mozhuli Actually containerd/containerd#849 is merged already, their api is changing so fast. :(

I'll still merge this PR first, and we could migrate to new containerd api and add new fake service later.

**Comment by mozhuli** [comment]
@Random-Liu OK! I will watch it

**Comment by mozhuli** [comment]
@Random-Liu update! I am sorry that I also use govender

File: vendor/github.com/opencontainers/image-spec/identity/chainid_test.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
+// Copyright 2016 The Linux Foundation
```
Comment (Random-Liu) [inline vendor/github.com/opencontainers/image-spec/identity/chainid_test.go]: I don't think godep vendors test.

Could you try godeps or manually remove this file?

**Comment by Random-Liu** [comment]
LGTM expect the final comment.

**Comment by Random-Liu** [comment]
Will merge after test passes. @mozhuli Thanks a lot! :)

File: vendor/github.com/opencontainers/image-spec/identity/chainid_test.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package identity
+
+import (
+	_ "crypto/sha256" // required to install sha256 digest support
```
Comment (mikebrow) [inline vendor/github.com/opencontainers/image-spec/identity/chainid_test.go]: ?

**Pull request merged**